proto/public/uber/cadence/api/v1/service_worker.proto

```diff
@@ -0,0 +1,323 @@
+  // fails to heartbeat within 'heartbeatTimeoutSeconds' interval for the ActivityTask, then it will be marked as
+  // timed out and 'ActivityTaskTimedOut' event will be written to the workflow history.
+  // Calling 'RecordActivityTaskHeartbeatByID' will fail with 'EntityNotExistsError' in such situations.  Instead of
+  // using 'taskToken' like in RecordActivityTaskHeartbeat, use Domain, WorkflowID and ActivityID.
+  rpc RecordActivityTaskHeartbeatByID(RecordActivityTaskHeartbeatByIDRequest) returns (RecordActivityTaskHeartbeatByIDResponse);
+
+
+  // RespondQueryTaskCompleted is called by application worker to complete a QueryTask (which is a DecisionTask for query)
+  // as a result of 'PollForDecisionTask' API call. Completing a QueryTask will unblock the client call to 'QueryWorkflow'
+  // API and return the query result to client as a response to 'QueryWorkflow' API call.
+  rpc RespondQueryTaskCompleted(RespondQueryTaskCompletedRequest) returns (RespondQueryTaskCompletedResponse);
+
+  // RequestCancelWorkflowExecution is called by application worker when it wants to request cancellation of a workflow
+  // instance. It will result in a new 'WorkflowExecutionCancelRequested' event being written to the workflow history
+  // and a new DecisionTask created for the workflow instance so new decisions could be made. It fails with
+  // 'EntityNotExistsError' if the workflow is not valid anymore due to completion or doesn't exist.
+  rpc RequestCancelWorkflowExecution(RequestCancelWorkflowExecutionRequest) returns (RequestCancelWorkflowExecutionResponse);
+}
+
+message PollForDecisionTaskRequest {
+  string domain = 1;
+  TaskList task_list = 2;
+  string identity = 3;
+  string binary_checksum = 4;
+}
+
+message PollForDecisionTaskResponse {
+  bytes task_token = 1;
+  WorkflowExecution workflow_execution = 2;
+  WorkflowType workflow_type = 3;
+  int64 previous_started_event_id = 4;
+  int64 started_event_id = 5;
+  int64 attempt = 6;
+  int64 backlog_count_hint = 7;
+  History history = 8;
+  bytes next_page_token = 9;
+  WorkflowQuery query = 10;
+  TaskList workflow_execution_task_list = 11;
+  google.protobuf.Timestamp scheduled_time = 12 [(gogoproto.stdtime) = true];
+  google.protobuf.Timestamp started_time = 13 [(gogoproto.stdtime) = true];
+  map<string, WorkflowQuery> queries = 14;
+}
+
+message RespondDecisionTaskCompletedRequest {
+  bytes task_token = 1;
+  repeated Decision decisions = 2;
+  bytes execution_context = 3;
+  string identity = 4;
+  StickyExecutionAttributes sticky_attributes = 5;
+  bool return_new_decision_task = 6;
+  bool force_create_new_decision_task = 7;
+  string binary_checksum = 8;
+  map<string, WorkflowQueryResult> query_results = 9;
+}
+
+message RespondDecisionTaskCompletedResponse {
+  PollForDecisionTaskResponse decision_task = 1;
+  map<string, ActivityLocalDispatchInfo> activities_to_dispatch_locally = 2;
```

Comment:
any reason this wouldn't be part of `PollForDecisionTaskResponse`?  granted it only saves 1 poll/respond cycle per process, but still kinda odd.

Reply:
Maybe @mkolodezny can answer this.

Reply:
Talked to @Groxx offline. This should be fine.

Reply:
after some reading + asking Qs of @meiliang86 : this makes sense!  tl;dr is:
- worker polls for a decision task, gets one
- decides it can/wants to execute some activities locally / eagerly
- adds those to `RespondDecisionTaskCompletedRequest`'s decisions, with a marker that it can run them locally
- server decides if / which ones should be executed locally, which is in this response

so it's not a field that makes sense on a poll, only on a response to a completed decision task.